src/Crypto/ExtCryptoCrypt.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types = 1);
+
+namespace Cmnty\Push\Crypto;
+
+use Cmnty\Push\Crypto\Exception\UnsupportedEncryptionMethod;
+use Crypto\Cipher as ExtCryptoCipher;
+
+class ExtCryptoCrypt implements Crypt
+{
+    /**
+     * Constructor.
+     *
+     * @throws UnsupportedEncryptionMethod When ext-crypto is not installed of the version is incompatible with the version of php.
+     */
+    public function __construct()
+    {
+        if (!phpversion('crypto')) {
```

Comment:
Never knew you could check extensions with phpversion. always used extension_loaded()